src/raiden_libs/cli.py

```diff
@@ -0,0 +1,73 @@
+import json
+import os
+from typing import Callable
+
+import click
+import structlog
+from eth_account import Account
+
+from raiden_libs.logging import setup_logging
+
+log = structlog.get_logger(__name__)
+
+
+def _open_keystore(ctx: click.Context, param: click.Parameter, value: str) -> None:
+    keystore_file = value
+    password = ctx.params.pop('password')
+    with open(keystore_file, 'r') as keystore:
+        try:
+            ctx.params['private_key'] = Account.decrypt(
+                keyfile_json=json.load(keystore),
+                password=password,
+            ).hex()
+        except ValueError as error:
+            log.critical(
+                'Could not decode keyfile with given password. Please try again.',
+                reason=str(error),
+            )
+            ctx.exit(1)
+
+
+def common_options(app_name: str) -> Callable:
+    """A decorator to be used with all service commands
+
+    It will pass two new args to the given func:
+    * private_key (as a result of `--keystore-file` and `--password`)
+    * state_db
+
+    The `app_name` will be used to determine the state_db location.
+    """
+    def decorator(func: Callable) -> Callable:
+        for option in reversed([
```

Comment:
Why use `reversed` here?

Reply:
So that the order or options is handled in the same way as when using them directly on the command.